### PR TITLE
add scss sourcempas to webpack + app weight analyzer

### PR DESCRIPTION
**Pull request by Druzzdil**

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -59,5 +59,6 @@ module.exports = {
       template: './src/index.ejs',
       favicon: './media/favicon.ico',
     }),
+    new BundleAnalyzerPlugin(),
```
Comment (patgrudniewski) [inline webpack.config.js]: add it to dev's webpack

**Review by patgrudniewski — CHANGES_REQUESTED** [review]

File: webpack.config.js (hunk 1 above)
Comment (Druzzdil) [reply webpack.config.js]: no problem

**Review by bartij — APPROVED** [review]

File: webpack.config.js [hunk 2]
```diff
@@ -1,63 +1,43 @@
...
+      plugins: [
+        "react-hot-loader/babel"
+      ],
+    };
+  }
+
+  return loader;
+}
+
+module.exports = require('./webpack.config.js');
```
Comment (patgrudniewski) [inline webpack.config.js]: why it is requiring itself?

File: webpack.dev.config.js [hunk 3]
```diff
@@ -1,41 +1,64 @@
...
-module.exports.module.loaders = module.exports.module.loaders.map(function cb(value) {
-  if (value.use instanceof Array) {
-    value.use = value.use.map(modifyBabelLoader);
-  } else if (value.use instanceof String) {
-    value.use = modifyBabelLoader(value.use);
-  }
-
-  return value;
-});
+const BundleAnalyzerPlugin = require('webpack-bundle-analyzer').BundleAnalyzerPlugin;
```
Comment (patgrudniewski) [inline webpack.dev.config.js]: disable autostart of it

File: webpack.config.js [hunk 4]
```diff
@@ -1,63 +1,43 @@
...
+
+module.exports.devServer.hot = true;
+
+module.exports.plugins.push(new webpack.HotModuleReplacementPlugin());
```
Comment (patgrudniewski) [inline webpack.config.js]: why non-dev webpack config contains `HotModuleReplacementPlugin`?

File: webpack.config.js [hunk 5]
```diff
@@ -1,63 +1,43 @@
-const path = require('path');
```
Comment (patgrudniewski) [inline webpack.config.js]: `npm run build` is not working

File: webpack.dev.config.js [hunk 6]
```diff
@@ -1,41 +1,64 @@
-const webpack = require("webpack");
+const path = require('path');
+const CleanWebpackPlugin = require('clean-webpack-plugin');
+const ExtractTextWebpackPlugin = require('extract-text-webpack-plugin');
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+const app_root = './src';
```
Comment (patgrudniewski) [inline webpack.dev.config.js]: add space between `includes` and `variables`

File: webpack.config.js (hunk 5 above)
Comment (patgrudniewski) [reply webpack.config.js]: `TypeError: Cannot set property 'hot' of undefined`

**Review by patgrudniewski — CHANGES_REQUESTED** [review]

File: webpack.dev.config.js (hunk 6 above)
Comment (Druzzdil) [reply webpack.dev.config.js]: it's not ok like it is ?

Comment (auamarto) [reply webpack.dev.config.js]: add empty line before app_root other lines are imports it's just visual thing but yes its not ok for us ;]

File: package.json [hunk 7]
```diff
@@ -77,7 +77,7 @@
     "sass-loader": "^6.0.6",
     "style-loader": "^0.13.1",
     "webpack": "^3.7.1",
-    "webpack-dev-server": "^2.9.2",
-    "whatwg-fetch": "^2.0.1"
```
Comment (patgrudniewski) [inline package.json:81]: remove it from `README.md`

File: package.json [hunk 8]
```diff
@@ -77,7 +77,7 @@
...
+    "webpack-bundle-analyzer": "^2.9.0",
```
Comment (patgrudniewski) [inline package.json:80]: add it to `README.md`

File: webpack.config.js [hunk 9]
```diff
@@ -1,63 +1,41 @@
...
+      plugins: [
+        "react-hot-loader/babel"
+      ],
+    };
+  }
+
+  return loader;
+}
+
+module.exports = require('./webpack.dev.config.js');
```
Comment (patgrudniewski) [inline webpack.config.js]: why base `webpack` is loading `dev` one?

File: webpack.config.js [hunk 10]
```diff
@@ -1,63 +1,41 @@
...
+
+module.exports.devServer.hot = true;
```
Comment (patgrudniewski) [inline webpack.config.js]: is this used on prod?

File: webpack.dev.config.js [hunk 11]
```diff
@@ -1,41 +1,66 @@
...
-    };
-  }
-
-  return loader;
-}
-
-module.exports = require('./webpack.config.js');
-
-module.exports.devServer.hot = true;
+const BundleAnalyzerPlugin = require('webpack-bundle-analyzer').BundleAnalyzerPlugin;
```
Comment (patgrudniewski) [inline webpack.dev.config.js]: require it [here](https://github.com/opstalent/React-Redux-Startekit/pull/4/files#diff-361932ce6b04b8880716e7995633922aR3)

**Review by patgrudniewski — CHANGES_REQUESTED** [review]

**Review by auamarto — COMMENTED** [review]
after README.md edit it's ok for me

**Review by auamarto — APPROVED** [review]

File: webpack.config.js [hunk 12]
```diff
@@ -59,5 +61,8 @@ module.exports = {
       template: './src/index.ejs',
       favicon: './media/favicon.ico',
     }),
+    new BundleAnalyzerPlugin({
```
Comment (patgrudniewski) [inline webpack.config.js:64]: is it needed for prod env?

**Pull request merged**